src/blk/kernel/KernelDevice.cc

```diff
@@ -133,8 +133,24 @@ int KernelDevice::open(const string& p)
   int r = 0, i = 0;
   dout(1) << __func__ << " path " << path << dendl;
 
+  struct stat statbuf;
+  bool is_block;
+  r = stat(path.c_str(), &statbuf);
+  if (r != 0) {
+    derr << __func__ << " stat got: " << cpp_strerror(r) << dendl;
+    goto out_fail;
+  }
+  is_block = (statbuf.st_mode & S_IFMT) == S_IFBLK;
   for (i = 0; i < WRITE_LIFE_MAX; i++) {
-    int fd = ::open(path.c_str(), O_RDWR | O_DIRECT);
+    int flags = 0;
+    if (lock_exclusive && is_block && (i == 0)) {
```

Comment:
It looks like `lock_exclusive` can be false when this is called through `BlueStore::_minimal_open_bluefs`.  Just checking that that is expected and that O_EXCL would still cover this case because -- I'm just conjecturing -- there would be another `BlockDevice`, this time without `set_no_exclusive_lock`, created later?

Reply:
Has this thread been followed up on?  Is there some assurance that in the "shared device" case, when `BlueStore::_minimal_open_bluefs` calls `BlueFS::add_block_device` in such a way that `lock_exclusive` gets cleared and `O_EXCL` isn't passed, we are still protected?

Reply:
@idryomov 
This PR does not revert existing advisory locking.
It is still possible to open BlueStore twice, but at least one of the opens must stem from ceph-bluestore-tool.
This minimizes possibility of accidental double opens.

The problem of hacky execution paths will be resolved by:
https://github.com/ceph/ceph/pull/47439
It is still active (not dropped) but waits for time to be continued.